core/src/instance.rs

```diff
@@ -438,7 +460,7 @@ pub mod tests {
 
         // @TODO don't use history length in tests
         // @see https://github.com/holochain/holochain-rust/issues/195
-        assert_eq!(instance.state().history.len(), 4);
+        assert_eq!(instance.state().history.len(), 5);
```

Comment:
haha, this pattern is sooo bad >.<

can't wait to clean these length checks out

Reply:
same :)

Reply:
Why is this needed anyway? Can it go since you've added the while-loop above that checks for the Commit action?